File: pset1/cash/cash.c
// program that asks the user for change and returns the minimum number of coins needed

#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;

    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    int numcoins = 0;
    int c = (int) round(change * 100);

    while (c >= 25)
    {
        numcoins++;
        c -= 25;
    }

    while (c >= 10)
    {
        numcoins++;
        c -= 10;
    }

    while (c >= 5)
    {
        numcoins++;
        c -= 5;
    }

    while (c >= 1)
    {
        numcoins++;
        c -= 1;
    }

    printf("%d\n", numcoins);
}

File: pset6/sentimental/cash/cash.py
from cs50 import get_float

if __name__ == "__main__":
    while True:
        cash = get_float("Change owed: ")
        cash *= 100
        if cash > 0:
            break
    coins = 0
    while cash >= 25:
        cash -= 25
        coins += 1
    while cash >= 10:
        cash -= 10
        coins += 1
    while cash >= 5:
        cash -= 5
        coins += 1
    while cash >= 1:
        cash -= 1
        coins += 1
    print(coins)